syntax = "proto3";

service ScheduleService {
    rpc GetListSchedules(Empty) returns (Schedules) {}
    rpc GetScheduleByDate(Date) returns (Movies) {}
    rpc AddSchedule(Schedule) returns (Schedule) {}
    rpc DeleteSchedule(Schedule) returns (Schedule) {}
}

message Date {
    int32 date = 1;
}

message Schedules {
    repeated Schedule schedules = 1;
}

message Schedule {
    int32 date = 1;
    repeated string movies = 2;
}

message Movies {
    repeated string movies = 1;
}

message Empty {}